pkg/api/vminfo.go

```diff
@@ -129,3 +130,54 @@ func ExtractVmInfo(vm *vmapi.VirtualMachine) (*VmInfo, error) {
 
 	return &info, nil
 }
+
+// the reason we have custom formatting for VmInfo is because without it, the formatting of memory
+// slot size (which is a resource.Quantity) has all the fields, meaning (a) it's hard to read, and
+// (b) there's a lot of unnecessary information. But in order to enable "proper" formatting given a
+// VmInfo, we have to implement Format from the top down, so we do.
+func (vm VmInfo) Format(state fmt.State, verb rune) {
+	switch {
+	case verb == 'v' && state.Flag('#'):
+		state.Write([]byte(fmt.Sprintf("api.VmInfo{Name:%q, Namespace:%q, Cpu:", vm.Name, vm.Namespace)))
+		vm.Cpu.format(state, verb)
+		state.Write([]byte(", Mem:"))
+		vm.Mem.format(state, verb)
+		state.Write([]byte(fmt.Sprintf(", AlwaysMigrate:%t, ScalingEnabled:%t}", vm.AlwaysMigrate, vm.ScalingEnabled)))
```

Comment:
I'm 100% sure that this doesn't matter, but it seems like we could maybe combine the number of calls here into one(ish?) sprintf?